sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -89,19 +93,158 @@ case class AnalyzeColumnCommand(
     // The first element in the result will be the overall row count, the following elements
     // will be structs containing all column stats.
     // The layout of each struct follows the layout of the ColumnStats.
-    val ndvMaxErr = sparkSession.sessionState.conf.ndvMaxError
     val expressions = Count(Literal(1)).toAggregateExpression() +:
-      attributesToAnalyze.map(ColumnStat.statExprs(_, ndvMaxErr))
+      attributesToAnalyze.map(statExprs(_, sparkSession.sessionState.conf))
```

Comment:
My feeling is that, we should run a job before calling `statExprs`, because `statExprs` need some extra information about buckets. I think it's better than hiding this job deep in `rowToColumnStats`.

Reply:
yea that'll make the logic clearer, thanks!